connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/SparkSessionE2ESuite.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+import org.scalatest.concurrent.Eventually._
+
+import org.apache.spark.sql.connect.client.util.RemoteSparkSession
+import org.apache.spark.util.ThreadUtils
+
+class SparkSessionE2ESuite extends RemoteSparkSession {
```

Comment:
Thank you for the investigation and suggestion, @LuciferYang .

If this is supposed to be a clean room unlike `ClientE2ETestSuite`. Could you add some test class description officially here? For example, something like `Do not include `org.apache.spark.sql.connect.client.*`?

>  import some classes from spark-connect-client-jvm.jar, such as org.apache.spark.sql.connect.client.SparkResult

Without an explicit written warning, this test suite could be broken again when someone adds a new test case. It will be a head-ache of re-analysis.

Reply:
Thanks @dongjoon-hyun ~ 

[b9e7e4a](https://github.com/apache/spark/pull/41487/commits/b9e7e4a615b447b48b91ed5d2de28332ed88285f) add waring comments to `SparkSessionE2ESuite`.

At the same time, I am preparing a pr use to add Maven test of connect related modules on GitHub Action(https://github.com/apache/spark/pull/41253), as Maven testing often failures before. If this pr can be merged, we can start testing the client module first.

Reply:
Non-blocking: I wonder if we can also add something to the class name to make it more indicative that it's a "clean room" suite? Maybe something like `SparkSessionCleanRoomE2ESuite` or some variation.

Reply:
Oh, sorry but let's not use `CleanRoom` keyword here, @vicennial . :)

Reply:
revert